azerothcore/azerothcore-wotlk
# feat(DB/Worldstates): added worldstates entry 20004 in characters DB

In `CharacterDatabaseCleaner.cpp` there are some piece of code that UPDATE the world state 20004  (`WS_CLEANING_FLAGS`) that currently does not exist.
In characters.worldstates there is 20005 named as `cleaning_flags` but it's wrong, I renamed it into `NextDailyQuestResetTime` and added the correct `cleaning_flags` entry.

##### CHANGES PROPOSED:
- added worldstate 20004 cleaning_flags
- renamed world state 20005 into `NextDailyQuestResetTime`

##### ISSUES ADDRESSED:
- Closes #2382

##### HOW TO TEST THE CHANGES:
- Add this PR to your core and import the sql query.
- Set `worldserver.conf` parameters correctly:
`CleanCharacterDB = 1`
`PersistentCharacterCleanFlags = 14`
- Set `worldstates` sql table value with the same value of `PersistentCharacterCleanFlags`, to make this execute the following query:
```sql
UPDATE `worldstates` SET `value`='14' WHERE `entry`='20004';
```
- log in in game and add dual talent specialization, learn some talents and spells
- check `character_talent` it will be full of data
- restart the worldserver, now `character_talent` will be empty, log in game and you will see your talents resetted
- change che worldserver.conf `PersistentCharacterCleanFlags=0`  and restart the server
- you will see that in `worldstates` **cleaning_flags** (20004) will be 0.

##### Target branch(es):
